platform/darwin/src/MGLTileSet.mm

```diff
@@ -0,0 +1,72 @@
+#import "MGLTileSet.h"
+
+#include <mbgl/util/tileset.hpp>
+
+@implementation MGLTileSet
+{
+    BOOL _minimumZoomLevelIsSet;
```

Comment:
If the minimum and maximum zoom levels are optional, then do they need to be represented as NSNumbers publicly? Otherwise, if you use the `-initWithTileURLTemplates:` initializer, these properties end up being 0.